src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotByRequestIdInterface.java

```diff
@@ -18,26 +19,26 @@ default StepResult prepareAndCreateSnapshot(
       FlightContext context,
       Snapshot snapshot,
       SnapshotRequestModel snapshotReq,
+      SnapshotService snapshotService,
       SnapshotBuilderService snapshotBuilderService,
-      SnapshotRequestDao snapshotRequestDao,
       SnapshotDao snapshotDao,
       AuthenticatedUserRequest userReq)
       throws InterruptedException {
-    var workingMap = context.getWorkingMap();
     UUID accessRequestId =
         snapshotReq.getContents().get(0).getRequestIdSpec().getSnapshotRequestId();
-    SnapshotAccessRequestResponse accessRequest = snapshotRequestDao.getById(accessRequestId);
+    SnapshotAccessRequestResponse accessRequest =
+        snapshotService.getSnapshotAccessRequestById(accessRequestId);
 
     UUID sourceSnapshotId = accessRequest.getSourceSnapshotId();
     Snapshot sourceSnapshot = snapshotDao.retrieveSnapshot(sourceSnapshotId);
+    Dataset dataset = sourceSnapshot.getSourceDataset();
 
-    AssetSpecification assetSpecification =
-        workingMap.get(
-            SnapshotWorkingMapKeys.SNAPSHOT_ASSET_SPECIFICATION, AssetSpecification.class);
+    AssetSpecification queryAssetSpecification =
```

Comment:
I remember you bringing up some concerns about regenerating the asset, such as if the settings change. Is there anything we need to do to mitigate those concerns?

Reply:
That's a great question. I think this is something we can think about in the future -- If we follow similar patterns to elsewhere in TDR, we could add a "locking" mechanism for the source snapshot so we don't update the settings during request snapshot creation.

Reply:
I've added a post-MVP ticket: https://broadworkbench.atlassian.net/browse/DC-1155

Reply:
In practice I think we should think of settings as being only set on snapshot creation. Right now we're still iterating on what the settings are, but if a snapshot's settings can change over time it could render a previous request invalid as it could change table mappings, etc. At some point in the future we remove the ability to set it later and only allow it to be set on snapshot creation.